src/hotel_reservations/types/project_config_types.py

```diff
@@ -0,0 +1,82 @@
+from typing import List, Literal, Optional, TypedDict, Union
+
+
+class Constraints(TypedDict, total=False):
+    """
+    Specifies optional constraints for a feature, such as minimum or maximum values.
+    - `min`: The minimum allowed value for the feature.
+    - `max`: The maximum allowed value for the feature.
+    """
+
+    min: Union[int, float]
+    max: Union[int, float]
+
+
+class NumFeature(TypedDict):
```

Comment:
you may consider to use pydantic instead. Pydantic enforces data types, while typing just provides hints (not enforced)

Reply:
Thanks for the suggestion, will implement Pydantic in the next PR. The addition of mypy to the pre-commit still checks the correctness of the types to a certain extent though, but Pydantic seems way more suitable for YAML